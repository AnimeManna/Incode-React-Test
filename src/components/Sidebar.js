import React, {Component} from 'react'

import {
    Input
} from 'semantic-ui-react'

class Sidebar extends Component{
    render(){
        return(
            <div>
                <Input icon="search" placeholder="Search" width="100%" />
            </div>
        )
    }
}

export default Sidebar