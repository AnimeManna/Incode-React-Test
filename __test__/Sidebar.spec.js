import React from 'react'
import configureStore from 'redux-mock-store'
import renderer from 'react-test-renderer'
import {shallow, render, mount} from 'enzyme'
import thunk from 'redux-thunk'

import Sidebar from '../src/components/Sidebar'
import toJson from "enzyme-to-json";

describe('Sidebar component', () => {
    const middelwares = [thunk]
    const mockStore = configureStore(middelwares)
    const initialState = {
        clientsReducer: {
            clients: [],
            user: {},
            viewClients: [
                {
                    general: {
                        firsName: 'Test Name',
                        lastName: 'Test LastName',
                        avatar: 'https://s3.amazonaws.com/uifaces/faces/twitter/andysolomon/128.jpg'
                    },
                    job: {
                        title: 'Test incorporation'
                    },
                    id: 0
                }
            ]

        }
    }
    const store = mockStore(initialState);
    describe('render', () => {
        test('View sidebar clients', () => {
            const wrapper = shallow(<Sidebar store={store}/>)
            const component = wrapper.dive();
            expect(toJson(component)).toMatchSnapshot();
        })
    })

})