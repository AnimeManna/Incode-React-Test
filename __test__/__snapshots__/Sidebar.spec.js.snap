// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar component render View sidebar clients 1`] = `
<Segment>
  <Input
    icon="search"
    onChange={[Function]}
    placeholder="Search"
    type="text"
  />
  <List
    style={
      Object {
        "maxHeight": 500,
        "minWidth": "100%",
        "overflow": "auto",
      }
    }
  >
    <Link
      key="0"
      replace={false}
      style={
        Object {
          "color": "black",
          "padding": 10,
        }
      }
      to="/client/0"
    >
      <ListItem
        onClick={[Function]}
        style={
          Object {
            "display": "flex",
          }
        }
      >
        <Image
          as="img"
          src="https://s3.amazonaws.com/uifaces/faces/twitter/andysolomon/128.jpg"
          style={
            Object {
              "height": 80,
              "width": 80,
            }
          }
          ui={true}
        />
        <ListContent>
          <ListHeader
            style={
              Object {
                "fontSize": 20,
                "padding": 5,
              }
            }
          >
             
             
            Test LastName
             
          </ListHeader>
          <ListDescription
            style={
              Object {
                "fontSize": 13,
                "padding": 5,
              }
            }
          >
             
            Test incorporation
             
          </ListDescription>
        </ListContent>
      </ListItem>
    </Link>
  </List>
</Segment>
`;
